binding/unity/Assets/Porcupine/PorcupineManager.cs

```diff
@@ -107,56 +105,51 @@ private void OnFrameCaptured(short[] pcm)
         /// Checks to see whether PorcupineManager is capturing audio or not
         /// </summary>
         /// <returns>whether PorcupineManager  is capturing audio or not</returns>
-        public bool IsRecording => _voiceProcessor.IsRecording;
+        public bool IsRecording => VoiceProcessor.Instance.IsRecording;
 
         /// <summary>
         /// Checks to see whether there are any audio capture devices available
         /// </summary>
         /// <returns>whether there are any audio capture devices available</returns>
         public bool IsAudioDeviceAvailable()
         {
-            _voiceProcessor.UpdateDevices();
-            return _voiceProcessor.CurrentDeviceIndex >= 0;
+            VoiceProcessor.Instance.UpdateDevices();
+            return VoiceProcessor.Instance.CurrentDeviceIndex >= 0;
         }
 
         /// <summary>
         /// Starts audio capture and wake word detection
         /// </summary>
         public void Start()
         {
-            if (_porcupine == null || _voiceProcessor == null)
+            if (_porcupine == null)
             {
                 throw new ObjectDisposedException("Porcupine", "Cannot start PorcupineManager - resources have already been released");
             }
-            _voiceProcessor.StartRecording(_porcupine.SampleRate, _porcupine.FrameLength);
+            VoiceProcessor.Instance.StartRecording(_porcupine.FrameLength, _porcupine.SampleRate);
         }
 
         /// <summary>
         /// Stops audio capture and wake word detection
         /// </summary>
         public void Stop()
         {
-            if (_porcupine == null || _voiceProcessor == null)
+            if (_porcupine == null)
             {
                 throw new ObjectDisposedException("Porcupine", "Stop called after PorcupineManager resources were released.");
             }
-            _voiceProcessor.StopRecording();
+            VoiceProcessor.Instance.StopRecording();
         }
 
         /// <summary>
         /// Free resources that were allocated to Porcupine and the voice processor
         /// </summary>
         public void Delete()
         {
-            if (_voiceProcessor != null)
+            VoiceProcessor.Instance.RemoveFrameListener(OnFrameCaptured);
+            if (VoiceProcessor.Instance.IsRecording)
             {
-                if (_voiceProcessor.IsRecording)
-                {
-                    _voiceProcessor.StopRecording();
-                }
-
-                _voiceProcessor.OnFrameCaptured -= OnFrameCaptured;
-                _voiceProcessor = null;
+                VoiceProcessor.Instance.StopRecording();
```

Comment:
the if should be if `numFrameListeners == 0` we call stop. If people are using separate managers, we only want to call stop if there are no other managers actively listening to the stream.